### PR TITLE
PYTHON-4879 Use shrub.py for compressor tests

### PR DESCRIPTION
**Pull request by blink1073**
Before: 288 tasks across 36 build variants
After: 78 tasks across 9 build variants

snappy compression RHEL8 py3 9

### CODE REVIEW

**Review by NoahStapp — APPROVED** [review]

File: .evergreen/scripts/generate_config.py [hunk 1]
```diff
@@ -337,8 +345,49 @@ def create_load_balancer_variants():
...
+    return variants
+
+
 ##################
 # Generate Config
 ##################
 
-generate_yaml(variants=create_load_balancer_variants())
+variants = create_compression_variants()
+# print(len(variants))
```
Comment (NoahStapp) [inline .evergreen/scripts/generate_config.py:392]:
```suggestion
```

**Review by NoahStapp — DISMISSED** [review]

File: .evergreen/scripts/generate_config.py (hunk 1 above)
Comment (NoahStapp) [reply .evergreen/scripts/generate_config.py:392]: Should we keep it in this final merge into master?

Comment (blink1073) [reply .evergreen/scripts/generate_config.py:392]: I'll just have to keep adding it back in every time while I'm building out these variants, doesn't seem worth doing

Comment (NoahStapp) [reply .evergreen/scripts/generate_config.py:392]: Can you let me know when the last shrub conversion PR is then? That way I know when any debugging code should be gone.

Comment (blink1073) [reply .evergreen/scripts/generate_config.py:392]: When I close the main ticket lol, a long way off

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**